ddtrace/internal/telemetry/writer.py

```diff
@@ -221,17 +216,11 @@ def app_heartbeat_event(self):
             #   add much value today.
             return
 
-        # DEV: Although the default flush interval is 60 seconds,
-        #   we want to explicitly control the flow here in case
-        #   the flush interval ever changes.
-        #
-        #   If the flush interval ever exceeds 60 seconds, then we
-        #   will queue a heartbeat event every flush:
-        #   60 <= X <= flush interval.
-        now = time.time()
-        if now - self._last_heartbeat >= TelemetryWriter.HEARTBEAT_MIN_INTERVAL:
-            self.add_event({}, "app-heartbeat")
-            self._last_heartbeat = now
+        if self._events_queue:
+            # Optimization: only queue heartbeat if no other events are queued
+            return
+
+        self.add_event({}, "app-heartbeat")
```

Comment:
We don't need to do this right? We send the heartbeat on each periodic call?

Reply:
I think the idea is that, if we don't have anything else to send, we just send the heartbeat?

Reply:
Yup but I had to make it as confusing as possible ;)